README.md

```diff
@@ -453,6 +453,73 @@ As you can see all the classes in a class hierarchy actually share oneclass vari
   end
 ```
 
+- Avoid explicit use of instance variables
+
+```ruby
+  # bad
+  class SomeClass
+    def initialize(foo)
+      @foo = foo
+    end
+  end
+  
+  def bar
+    @foo == "Foo"
+  end
+  
+  # good
+  class SomeClass
+    attr_reader :foo
+    
+    def initialize(foo)
+      @foo = foo
+    end
+    
+    def bar
+      foo == "Foo"
+    end
+  end
+  
+  # better
+  class SomeClass
+    def initialize(foo)
+      @foo = foo
+    end
+    
+    def bar
+      foo == "Foo"
+    end
+    
+    private
+    
+    attr_reader :foo
+    
+  end
+```
+
+- Avoid complex logic in the initialiser
```

Comment:
Well, in some cases I guess it's better to let the program crash while doing initialising, rather than doing a bunch of heavy lifting and then realise that the record does not exist.